tests/test_validator.py

```diff
@@ -259,9 +260,168 @@ async def test_verify_there_is_not_example_repetition_in_intents():
 
 
+async def test_verify_actions_in_stories_not_in_domain(
+    tmp_path: Path, domain_path: Text
+):
+    story_file_name = tmp_path / "stories.yml"
+    story_file_name.write_text(
+        """
+        version: "2.0"
+        stories:
+        - story: story path 1
+          steps:
+          - intent: greet
+          - action: action_test_1
+        """
+    )
+
+    importer = RasaFileImporter(
+        domain_path=domain_path, training_data_paths=[story_file_name],
+    )
+    validator = await Validator.from_importer(importer)
+    with pytest.raises(ActionNotFoundException) as e:
+        validator.verify_actions_in_stories_rules()
+
+    assert "Cannot access action 'action_test_1'" in str(e.value)
+
+
+async def test_verify_actions_in_rules_not_in_domain(tmp_path: Path, domain_path: Text):
+    rules_file_name = tmp_path / "rules.yml"
+    rules_file_name.write_text(
+        """
+        version: "2.0"
+        rules:
+        - rule: rule path 1
+          steps:
+          - intent: goodbye
+          - action: action_test_2
+        """
+    )
+    importer = RasaFileImporter(
+        domain_path=domain_path, training_data_paths=[rules_file_name],
+    )
+    validator = await Validator.from_importer(importer)
+    with pytest.raises(ActionNotFoundException) as e:
+        validator.verify_actions_in_stories_rules()
+
+    assert "Cannot access action 'action_test_2'" in str(e.value)
+
+
+async def test_verify_form_slots_invalid_domain(tmp_path: Path):
+    domain = tmp_path / "domain.yml"
+    domain.write_text(
+        """
+        version: "2.0"
+        forms:
+          name_form:
+             first_name:
+             - type: from_text
+             last_name:
+             - type: from_text
+        slots:
+             first_name:
+                type: text
+             last_nam:
+                type: text
+        """
+    )
+    importer = RasaFileImporter(domain_path=domain)
+    validator = await Validator.from_importer(importer)
+    with pytest.warns(UserWarning) as w:
+        validity = validator.verify_form_slots()
+        assert validity is False
+
+    assert (
+        w[0].message.args[0]
+        == "The form slot 'last_name' in form 'name_form' is not present in the domain slots."
+        "Please add the correct slot or check for typos."
+    )
+
+
 async def test_invalid_domain_mapping_policy():
     importer = RasaFileImporter(
         domain_path="data/test_domains/default_with_mapping.yml"
     )
     validator = await Validator.from_importer(importer)
     assert validator.verify_domain_validity() is False
+
+
+async def test_valid_stories_actions_in_domain(tmp_path: Path):
+    domain = tmp_path / "domain.yml"
+    domain.write_text(
+        """
+        version: "2.0"
+        intents:
+        - greet
+        actions:
+        - action_greet
+        """
+    )
+    story_file_name = tmp_path / "stories.yml"
+    story_file_name.write_text(
```

Comment:
an example with a default action like `action_restart` would be good as well I think